salt/utils/event.py

```diff
@@ -401,9 +422,10 @@ def _get_event(self, wait, tag, match_func=None):
                 if socks.get(self.sub) != zmq.POLLIN:
                     continue
 
-                # Please do not use non-blocking mode here. Reliability is
-                # more important than pure speed on the event bus.
-                ret = self.get_event_block()
+                if no_block is True:
```

Comment:
I think when I tried it, 0 is blocking (no timeout) in `poll`
It just needs the `if no_block is True` to be before the `while`.
It should then loop `get_event_noblock()`, matching it against tags (like it does inside the `while`) until it finds a match or returns `None`